src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,27 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.excpt.NotValidUserData;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new NotValidUserData("There is already a user with the given name");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new NotValidUserData("User login must have at least 6 characters");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new NotValidUserData("User password must have at least 6 characters");
+        }
```

Comment:
Consider adding a null check for `user.getPassword()` to avoid potential `NullPointerException`.